packages/common/src/types/share.ts

```diff
@@ -1,12 +1,37 @@
+/**
```

Comment:
should we put API related typing in a separate folder to make it explicit? `types/api/share.ts` maybe?